src/cycax/cycad/engine_openscad.py

```diff
@@ -0,0 +1,209 @@
+
+        Args:
+            lookup: this will be the dictionary that contains the details about the cube so that is can be encoded in scad.
+
+        """
+        center = ""
+        if lookup["center"] is True:
+            center = ", center=true"
+        res = (
+            "cube(["
+            + str(lookup["x_size"])
+            + ", "
+            + str(lookup["y_size"])
+            + ","
+            + str(lookup["z_size"])
+            + "]"
+            + center
+            + ");"
+        )
+        return res
+
+    def decode_external(self, data_file: str) -> str:
+        """
+        This method will return the scad string necessary for processing the external part.
+
+        Args:
+            data_file : This will contain the part number of the external part so that it can be fetched from the parts_stl folder.
+
+        """
+
+        return 'import("../parts_stl/' + data_file + '.stl");'
+
+    def decode_hole(self, lookup: dict) -> str:
+        """
+        This method will return the string that will have the scad for a hole.
+
+        Args:
+            lookup : This will be a dictionary containing the necessary information about the hole.
+
+        """
+        tempdiam = lookup["diameter"] / 2
+        res = "cylinder(r=" + str(tempdiam) + ", h=" + str(lookup["depth"]) + ", $fn=64);"
+        return res
+
+    def decode_nut(self, lookup: dict) -> str:
+        """
+        This method will return the string that will have the scad for a nut cut out.
+
+        Args:
+            lookup : This will be a dictionary containing the necessary information about the nut.
+
+        """
+        res = (
+            "rotate(["
+            + str(0)
+            + ", "
+            + str(0)
+            + ", "
+            + str(0)
+            + "])"
+            + "cylinder(r="
+            + str(lookup["nut_type"])
+            + ", h="
+            + str(lookup["depth"])
+            + ", $fn=6);"
+        )
+        return res
+
+    def decode_cut(self) -> str:
+        """
+        This method returns a simple OpenSCAD string neceseray to cut.
+        """
+        return "difference(){"
+
+    def translate(self, lookup: dict) -> str:
+        """
+        This will move the object around and return the scad necessary.
+
+        Args:
+            lookup : This will be a dictionary containing the necessary information about the hole.
+        """
+        res = "translate([" + str(lookup["x"]) + ", " + str(lookup["y"]) + ", " + str(lookup["z"]) + "])"
+        return res
+
+    def rotate(self, side: str) -> str:
+        """
+        This will rotate the object and return the scad necessary.
+
+        ???Would it make sense to also have a dictionary here similar to location swap methods???
+
+        Args:
+            side : this is the side as retrieved form the dictionary.
+        """
+        if side == TOP:
+            return "rotate([" + str(0) + ", " + str(180) + ", " + str(0) + "])"
```

Comment:
Is this not the same as:
`return "rotate([0, 180, 0])"`

Then your question from line 107 is Yes, dictionary would be a lot simpler to read and understand.